hpcgap/lib/oper.g

```diff
@@ -1276,6 +1276,106 @@ BIND_GLOBAL( "DeclareAttribute", function ( arg )
 end );
 
 
+##############################################################################
+##
+##  This code adds thread-local attributes, which are used in HPC-GAP.
+##  These are not complete replacements for normal attributes.
+##  In particular, they are not filters, so cannot be used in method selection.
+##
+##  The main problem with thread-local attributes is that only a subset of the
+##  threads may have a value for a particular mutable attribute, but filters
+##  are not stored thread-locally. Therefore thread-local attributes are not
+##  a perfect replacement for traditional attributes.
+##
+##  The reason we have thread-local attributes is for code which uses
+##  mutable attribute as a general scratch-space, like StabChainMutable, which
+##  cannot be shared between threads.
```

Comment:
This paragraph sounds very strange... Can you really use "The reason ..." like this? Perhaps "The reason ... is to allow code ...". Also, this explanation left me a bit confused.... why are mutable attributes suddenly mention? Are TLS attributes meant to replace them? Or can TLS attributes come in mutable and immutable versions? Oh wait, an immutable attribute probably never needs to be TLS... so perhaps TLS attributes are a variation of mutable attributes then? But then, why doesn't this whole documentation comment lead with that quite central piece of information?

Ignoring all that for the moment, I feel this might be a possible substitute paragraph: "Despite these limitations, TLS attributes are useful for certain applications. For example, they can be used to provide scratch-space for things like StabChainMutable, which cannot [must not???] be shared between threads."